src/main/java/com/synopsys/integration/detect/lifecycle/run/operation/OperationRunner.java

```diff
@@ -464,14 +466,20 @@ public final void publishRapidResults(File jsonFile, RapidScanResultSummary summ
      */
     public void generateComponentLocationAnalysisIfEnabled(BdioResult bdio) throws OperationException {
         if (detectConfigurationFactory.isComponentLocationAnalysisEnabled()) {
-            if (!bdio.getCodeLocationNamesResult().getCodeLocationNames().isEmpty()) {
-                auditLog.namedPublic(
-                        "Generate Component Location Analysis File for All Components",
-                        () -> new GenerateComponentLocationAnalysisOperation().locateComponentsForOfflineDetectorScan(bdio, directoryManager.getScanOutputDirectory(), directoryManager.getSourceDirectory())
-                );
-            } else {
+            if (applicableDetectorsIncludeAtLeastOneSupportedDetector(bdio.getApplicableDetectorTypes())) {
+                if (!bdio.getCodeLocationNamesResult().getCodeLocationNames().isEmpty()) {
```

Comment:
Should both be checked for null?

Reply:
Back tracing through the code, there is no execution path that allows the BdioResult at this step to have null instance variables. So `bdio.getApplicableDetectors()` and `bdio.getCodeLocationNamesResult().getCodeLocationNames()` will always return at least an empty list or set.